helm-charts/stable/amazon-eks-pod-identity-webhook/config/job.sh

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -x
+
+echo "Running job.sh in $(pwd)"
+
+yum install -y jq
+
+KUBECTL_VERSION=v1.18.9
+curl -sSL "https://distro.eks.amazonaws.com/kubernetes-1-18/releases/1/artifacts/kubernetes/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" -o /bin/kubectl
+chmod +x /bin/kubectl
+
+secret_name=$(kubectl get sa default -o jsonpath='{.secrets[0].name}')
+CA_BUNDLE=$(kubectl get secret/$secret_name -o jsonpath='{.data.ca\.crt}' | tr -d '\n')
+cat /config/mutatingwebhook.yaml | sed -e "s|\${CA_BUNDLE}|${CA_BUNDLE}|g" > mutatingwebhook.yaml
+kubectl apply -f mutatingwebhook.yaml
+
+# Loop for a total of 50 seconds to give time for webhook to create CertificateSigningRequest
+for i in {1..10}; do
+    for c in $(kubectl get csr -o json | jq -r '.items[] | select(.spec.username=="system:serviceaccount:default:pod-identity-webhook" and .status=={}).metadata.name'); do
```

Comment:
Should we consider moving this script as part of the configmap? The namespace is configurable when you install a chart. 

There might be a way to hack this so that the namespace is configurable in this script

Reply:
This does get templated into the ConfigMap. Do you mean just defining the script in the ConfigMap file? I thought it was cleaner to have it as a separate file, but I agree that it would be nice to make the namespace configurable.

Reply:
The hack could be as simple as:

```bash
NAMESPACE=$(cat /var/run/secrets/kubernetes.io/namespace)
for i in {1..10}; do
    for c in $(kubectl get csr -o json | jq -r '.items[] | select(.spec.username=="system:serviceaccount:$NAMESPACE:pod-identity-webhook" and .status=={}).metadata.name'); do
```

Reply:
Good suggestion!

Reply:
The cat didn't seem to work, however I planned on templating out the webhook name as well, so adding another env var that references metadata.namespace instead

Reply:
Oh it didnt work because the path is `/var/run/secrets/kubernetes.io/serviceaccount/namespace`, but I will still just pass it in as an env var. We wouldn't need the extra permissions after then.